consensus/istanbul/core/roundstate.go

```diff
@@ -141,51 +149,36 @@ func (s *roundState) Sequence() *big.Int {
 	return s.sequence
 }
 
-func (s *roundState) SetPreparedCertificate(preparedCertificate istanbul.PreparedCertificate) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-
-	s.preparedCertificate = preparedCertificate
-}
-
 func (s *roundState) CreateAndSetPreparedCertificate(quorumSize int) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	prepareOrCommitSize := s.getPrepareOrCommitSize()
-	if prepareOrCommitSize >= quorumSize {
-		messages := make([]istanbul.Message, prepareOrCommitSize)
-		i := 0
-		for _, message := range s.Prepares.Values() {
-			messages[i] = *message
-			i++
+	messages := make([]istanbul.Message, quorumSize)
+	i := 0
+	for _, message := range s.Prepares.Values() {
+		if i == quorumSize {
+			break
 		}
-		for _, message := range s.Commits.Values() {
-			if s.Prepares.Get(message.Address) == nil {
-				messages[i] = *message
-				i++
-			}
+		messages[i] = *message
+		i++
+	}
+	for _, message := range s.Commits.Values() {
+		if i == quorumSize {
+			break
 		}
-		s.preparedCertificate = istanbul.PreparedCertificate{
-			Proposal:                s.Preprepare.Proposal,
-			PrepareOrCommitMessages: messages,
+		if s.Prepares.Get(message.Address) == nil {
+			messages[i] = *message
+			i++
 		}
-		return nil
-	} else {
+	}
+	if i != quorumSize {
```

Comment:
Question: Why do we need to `Ensure we never build a PreparedCertificate larger than quorumSize`?

Reply:
It's not strictly necessary